dartdoc_test/example/example.dart

```diff
@@ -0,0 +1,109 @@
+/// Example of documentation comments and code samples in Dart.
+
+/// This is a simple class example.
+///
+/// This class demonstrates a simple Dart class with a single method.
+class SimpleClass {
+  /// Adds two numbers together.
+  ///
+  /// The method takes two integers [a] and [b], and returns their sum.
+  ///
+  /// Example:
+  /// ```dart
+  /// final result = SimpleClass().add(2, 3);
+  /// print(result); // 5
+  /// ```
+  int add(int a, int b) {
+    return a + b;
+  }
+}
+
+/// This class demonstrates more complex documentation comments,
+/// including parameter descriptions and a detailed method explanation.
+class ComplexClass {
+  /// Multiplies two numbers together.
+  ///
+  /// Takes two integers [x] and [y], multiplies them and returns the result.
+  /// This method handles large integers and ensures no overflow occurs.
+  ///
+  /// Example:
+  /// ```dart
+  /// final product = ComplexClass().multiply(4, 5);
+  /// print(product); // 20
+  /// ```
+  int multiply(int x, int y) {
+    return x * y;
+  }
+
+  /// Calculates the factorial of a number.
+  ///
+  /// This method uses a recursive approach to calculate the factorial of [n].
+  /// It throws an [ArgumentError] if [n] is negative.
+  ///
+  /// Example:
+  /// ```dart
+  /// final fact = ComplexClass().factorial(5);
+  /// print(fact); // 120
+  /// ```
+  ///
+  /// Throws:
+  /// - [ArgumentError] if [n] is negative.
+  int factorial(int n) {
+    if (n < 0) throw ArgumentError('Negative numbers are not allowed.');
+    return n == 0 ? 1 : n * factorial(n - 1);
+  }
+}
+
+/// Utility class for various static methods.
+class Utility {
```

Comment:
I know this is an example, but this kind of class is rather uncommon in dart -- or rather it ought to be :see_no_evil: 

static functions should just be top-level, if there is no reason for them to be static.

It might be more interesting to just make `gcd` and `isPalindrome` top-level functions.

If we want something fun with classes we should try to make subclass instead :D

Reply:
thank you for reviewing. I fix it.